### PR TITLE
feat: add check-learn-dream precheck

### PR DESCRIPTION
**Pull request by dmcaulay**
## Summary
- Add `check-learn-dream.ts` precheck for the learn-dream cron, using the same anchor-file pattern as `check-docs-freshness.ts`
- Reads `state/learn-dream-last-run.json` for a last-run timestamp and compares it against session transcript mtimes under `~/.claude/projects/<sanitized-workspace-path>/*.jsonl`, exiting 1 when nothing is newer (skip the cron) and 0 otherwise (permissive on first run or any read failure)
- Refresh `docs/architecture.md` to reflect the new precheck script

## Acceptance Criteria
| # | Criterion | Status | Evidence |
|---|-----------|--------|----------|
| 1 | Reads last-run anchor from state/learn-dream-last-run.json (absent -> exit 0) | MET | check-learn-dream.ts reads anchor; exits 0 permissively when null |
| 2 | Exits 1 when no transcript newer than anchor | MET | newerCount===0 -> exit 1, empty output |
| 3 | Exits 0 with summary when at least one transcript newer | MET | exit 0 with count summary |
| 4 | check-learn-dream.test.ts, deps-injection, sibling naming, covers scenarios | MET | 6 tests covering no-anchor, anchor-newer-than-all, empty-list, anchor-older-than-one, count-summary, and permissive-null-read |

## Test Plan
- `bun test plugins/shipwright/scripts/check-learn-dream.test.ts` — 6/6 pass
- `bunx tsc --noEmit` (plugins/shipwright) — clean
- `bunx biome lint .` — clean
- Full `bun test` — 3265 pass, 15 pre-existing failures (confirmed identical on `main`, unrelated to this change: kubernetes-client, plugin installation, chat-token-reporter)
- [x] Pre-commit checks passing

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/shipwright/scripts/check-learn-dream.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+): string {
+  return workspacePath.replace(/\//g, "-");
+}
+
+function buildProductionDeps(): Deps {
+  const cwd = process.cwd();
+
+  return {
+    readLastRunAnchor: (): string | null => {
+      const anchorPath = join(cwd, "state", "learn-dream-last-run.json");
```
Comment (dodizzle) [inline plugins/shipwright/scripts/check-learn-dream.ts:106]: The anchor path is read here, but nothing in the codebase writes `state/learn-dream-last-run.json`. The `learn-dream` command and `learning-dreamer` agent have no step that creates or updates this file. Because the file never exists, `readLastRunAnchor()` always returns `null`, and the precheck always exits 0 (first-run permissive path) — every cron firing still reaches the full LLM session. Add a write step to whichever command/agent runs the dream job (e.g. after the learning-dreamer agent completes, write `{ "lastRun": new Date().toISOString() }` to this path).

**Review by dodizzle — COMMENTED** [review]
COMMENT — The core logic is correct and the tests are solid, but `state/learn-dream-last-run.json` is never written by any skill (`learn-dream` command and `learning-dreamer` agent have no step that writes this file). The precheck will always hit the `null` → exit 0 path, so every cron firing still reaches the full LLM session — the guard provides no protection until something writes the anchor. Either add a write step to the `learn-dream` command / `learning-dreamer` agent, or note the gap so it can be tracked.

File: plugins/shipwright/scripts/check-learn-dream.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+/**
+ * Derive the Claude Code project directory name for a workspace path, the
+ * same way Claude Code itself does: the absolute path with `/` replaced by
+ * `-` (e.g. `/data/agent-home/workspace` → `-data-agent-home-workspace`).
+ */
+export function sanitizeWorkspacePathForClaudeProjects(
+  workspacePath: string,
+): string {
+  return workspacePath.replace(/\//g, "-");
```
Comment (dodizzle) [inline plugins/shipwright/scripts/check-learn-dream.ts]: The Claude Code project directory encoding rule replaces both `/` and `.` with `-` (per `plugins/shipwright/skills/investigate-cron/SKILL.md` line 55), but this only replaces `/`. A workspace path with dots (e.g. `/home/user/my.workspace`) produces the wrong directory name, causing `listTranscriptMtimes` to find nothing and return `null` — silently falling back to permissive exit 0 on every firing.

```ts
return workspacePath.replace(/[/\.]/g, "-");
```

**Review by dodizzle — COMMENTED** [review]
COMMENT — one bug in `sanitizeWorkspacePathForClaudeProjects`: it only replaces `/` with `-` but the Claude Code encoding rule (per `investigate-cron/SKILL.md` line 55) requires replacing both `/` and `.` with `-`. On any workspace path containing dots this will silently produce the wrong transcript directory and fall back to permissive exit 0 on every cron firing. Fix: use a regex that matches both `/` and `.`. Everything else looks good — clean deps-injection design, solid test coverage, well-documented anchor write contract.

**Review by dmcaulay — COMMENTED** [review]
APPROVE — Both prior findings are correctly resolved: `sanitizeWorkspacePathForClaudeProjects` now replaces both `/` and `.` (matching `investigate-cron/SKILL.md`'s documented encoding rule), with new tests covering dotted paths; and `learn-dream.md` now has a step instructing the command to write `state/learn-dream-last-run.json` on successful completion, ordered to avoid falsely advancing the anchor on partial failures. 9/9 tests pass locally, lint clean, CI green. No new issues found.

**Review by dodizzle — COMMENTED** [review]
**Re-review after a19d8fa** — prior finding (dot-to-dash encoding in `sanitizeWorkspacePathForClaudeProjects`) has been addressed.

**New finding: precheck script not wired into the cron definition**

The learn-dream entry in `admin/src/system-crons.ts` (line 76) has no `preCheck` field, so `check-learn-dream.ts` is never invoked by the cron system. Every other cron with a precheck script declares it via `preCheck` (e.g. `check-docs-freshness.ts` at line 72 just above). Without this wiring, the cron fires unconditionally — the precheck script exists and is tested, but has no effect.

Fix: add `preCheck: "shipwright:check-learn-dream.ts"` to the learn-dream cron entry:
```ts
{
  name: "learn-dream",
  schedule: "0 3 * * *",
  preCheck: "shipwright:check-learn-dream.ts",
  prompt: "/shipwright:learn-dream --since 1d --review",
  silent: true,
  enabled: false,
}
```

**COMMENT** — one wiring step missing before this PR achieves its stated purpose.

**Review by dodizzle — COMMENTED** [review]
**Re-review after a19d8fa** — prior finding (dot-to-dash encoding in `sanitizeWorkspacePathForClaudeProjects`) has been addressed.

**New finding: precheck script not wired into the cron definition**

The learn-dream entry in `admin/src/system-crons.ts` (line 76) has no `preCheck` field, so `check-learn-dream.ts` is never invoked by the cron system. Every other cron with a precheck script declares it via `preCheck` (e.g. `check-docs-freshness.ts` at line 72 just above). Without this wiring, the cron fires unconditionally — the precheck script exists and is tested, but has no effect.

Fix: add `preCheck: "shipwright:check-learn-dream.ts"` to the learn-dream cron entry:
```ts
{
  name: "learn-dream",
  schedule: "0 3 * * *",
  preCheck: "shipwright:check-learn-dream.ts",
  prompt: "/shipwright:learn-dream --since 1d --review",
  silent: true,
  enabled: false,
}
```

**COMMENT** — one wiring step missing before this PR achieves its stated purpose.

**Review by dodizzle — APPROVED** [review]
All prior findings resolved. APPROVE.

**Prior findings:**
- Dot-encoding bug in `sanitizeWorkspacePathForClaudeProjects` — fixed (regex now `/[/.]/g`)
- `preCheck: "shipwright:check-learn-dream.ts"` missing from learn-dream cron — fixed (added at `admin/src/system-crons.ts:78`)

Clean implementation following established precheck patterns. Tests comprehensive (6 unit tests, all scenarios covered), CI green.

**Review by dodizzle — COMMENTED** [review]
**Re-review after 1f2b676** — prior findings (dot-encoding, preCheck wiring) are confirmed resolved.

**New finding: permissive fallback paths suppress the cron instead of running it**

Two fallback paths in `check-learn-dream.ts` return `{ exit: 0, output: "" }`: when `listTranscriptMtimes()` is `null` (read failure, line 54) and when `anchorMs` is `NaN` (unparsable anchor, line 60). In `cron-handler.ts:253`, the handler evaluates `if (checkExitCode !== 0 || !output)` after `output = checkOutput.trim()` — empty string trims to `""`, so `!output` is `true`, and the session is suppressed with `"preCheck:no-work"`. The comments say "unknown state, exit permissively" but the actual effect is the opposite: the job doesn't run.

**Fix**: Return a non-empty output string in both fallback branches:
```ts
// line 54:
return { exit: 0, output: "Transcript read failed — running permissively." };
// line 60:
return { exit: 0, output: "Anchor unparsable — running permissively." };
```

**COMMENT** — one fix needed before merge.

File: plugins/shipwright/scripts/check-learn-dream.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  if (lastRun === null) {
+    return {
+      exit: 0,
+      output: "No last-run anchor found — running full learn-dream check.",
+    };
+  }
+
+  const anchorMs = Date.parse(lastRun);
+
+  const mtimes = deps.listTranscriptMtimes();
```
Comment (dodizzle) [inline plugins/shipwright/scripts/check-learn-dream.ts:55]: This returns empty `output` on exit 0. `cron-handler.ts:253` evaluates `if (checkExitCode !== 0 || !output)` — empty string trims to `""`, so `!output` is `true` and the cron session is suppressed (not run permissively as intended).

Fix: `return { exit: 0, output: "Transcript read failed — running permissively." };`

File: plugins/shipwright/scripts/check-learn-dream.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+  // Read failure — unknown state, exit permissively
+  if (mtimes === null) {
+    return { exit: 0, output: "" };
+  }
+
```
Comment (dodizzle) [inline plugins/shipwright/scripts/check-learn-dream.ts:64]: Same issue — empty output on exit 0 causes the cron handler to suppress the session rather than run permissively.

Fix: `return { exit: 0, output: "Anchor unparsable — running permissively." };`

**Review by dodizzle — APPROVED** [review]
APPROVE — all prior findings resolved.

**Prior findings:**
- Dot-encoding bug in `sanitizeWorkspacePathForClaudeProjects` — fixed (regex `/[/.]/g`)
- `preCheck: "shipwright:check-learn-dream.ts"` missing from learn-dream cron — fixed
- Permissive fallback paths returning `{ exit: 0, output: "" }` suppressing the cron — fixed; both branches now return descriptive non-empty output strings

Test coverage is comprehensive: 8 tests covering no-anchor, anchor-newer-than-all, empty-list, anchor-older-than-one, count-summary, permissive-null-read, unparsable-anchor, and path sanitization with slash/dot/mixed patterns. CI green.
